src/internal/clusterstate/v2.5.0/util.go

```diff
@@ -0,0 +1,57 @@
+package v2_5_0
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"go.uber.org/zap"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+)
+
+type postgresBatcher struct {
+	ctx   context.Context
+	tx    *pachsql.Tx
+	stmts []string
```

Comment:
Rather than `[]string`, this would be an improvement:

```go
type statement struct {
  query string
  args []any
}

type postgresBatcher struct {
  tx *pachsql.Tx
  stmts []statement
}
⋮
func (pb *postgresBatcher) execute(ctx context.Context) error {
  ⋮
  var (
    queries = make([]string, len(pb.stmts)
    args []any
  )
  for i, stmt := range pb.stmts {
    queries[i] = stmt.query
    args = append(args, stmt.args...)
  }
  ⋮
  if _, err := pb.tx.ExecContext(ctx, strings.Join(queries, ";"), args...) { … }
```

The issue is that constructing SQL queries by raw string interpolation exposes one to SQL injection. I think it might even make the caller side somewhat more legible, since one can have a `map[string]any` for the update fields.